File: Algorithms/src/2946. Matrix Similarity After Cyclic Shifts/Code/solution.java
class Solution {
    public boolean areSimilar(int[][] mat, int k) {
        int m = mat.length;
        int n = mat[0].length;
        
        k %= n;
        
        for(int i = 0; i < m; i++) {
            for(int j = 0; j < n; j++) {
                int newCol;
                
                if(i % 2 == 0) {
                    newCol = (j + k) % n;
                } else {
                    newCol = (j - k + n) % n;
                }
                
                if(mat[i][j] != mat[i][newCol]) {
                    return false;
                }
            }
        }
        return true;
    }
}

File: Algorithms/src/2946. Matrix Similarity After Cyclic Shifts/Code/solution.js
var areSimilar = function (mat, k) {
  let m = mat.length;
  let n = mat[0].length;

  k = k % n;

  for (let i = 0; i < m; i++) {
    for (let j = 0; j < n; j++) {
      let newCol;

      if (i % 2 === 0) {
        newCol = (j + k) % n;
      } else {
        newCol = (j - k + n) % n;
      }

      if (mat[i][j] !== mat[i][newCol]) {
        return false;
      }
    }
  }

  return true;
};
